compiler/compiler_test.go

```diff
@@ -44,12 +44,12 @@ func TestCompileAndEval(t *testing.T) {
 			fn:   `(r) => "n = ${r.n}"`,
```

Comment:
Can you add a test case that shows interpolation working for other Stringable types besides string?